src/interface/make_swarm_pack_descriptor.hpp

```diff
@@ -0,0 +1,59 @@
+//========================================================================================
+// (C) (or copyright) 2020-2023. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef INTERFACE_MAKE_SWARM_PACK_DESCRIPTOR_HPP_
+#define INTERFACE_MAKE_SWARM_PACK_DESCRIPTOR_HPP_
+
+#include <algorithm>
+#include <functional>
+#include <limits>
+#include <map>
+#include <memory>
+#include <regex>
+#include <set>
+#include <string>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "interface/mesh_data.hpp"
+#include "interface/meshblock_data.hpp"
+#include "interface/metadata.hpp"
+#include "interface/state_descriptor.hpp"
+#include "interface/swarm_pack.hpp"
+#include "interface/variable.hpp"
+#include "mesh/mesh.hpp"
+
+namespace parthenon {
+
+template <typename TYPE>
+inline auto MakeSwarmPackDescriptor(const std::string &swarm_name,
+                                    const std::vector<std::string> &vars) {
+  impl::SwarmPackDescriptor<TYPE> base_desc(swarm_name, vars);
+  return typename SwarmPack<TYPE>::Descriptor(base_desc);
+}
+
+template <typename TYPE, class... Ts>
+inline auto MakeSwarmPackDescriptor(const std::string &swarm_name) {
+  static_assert(sizeof...(Ts) > 0, "Must have at least one variable type for type pack");
+
+  std::vector<std::string> vars{Ts::name()...};
+
+  return typename SwarmPack<TYPE, Ts...>::Descriptor(
+      static_cast<impl::SwarmPackDescriptor<TYPE>>(
+          MakeSwarmPackDescriptor<TYPE>(swarm_name, vars)));
+}
```

Comment:
I think because you have the `PackDescriptor` you can infer `TYPE` from `Ts` in only this function so you can do the automatic type inferrence we discussed. And I *think* only this function needs to change. Maybe this is sufficient?
```suggestion
template <class... Ts>
inline auto MakeSwarmPackDescriptor(const std::string &swarm_name) {
  static_assert(sizeof...(Ts) > 0, "Must have at least one variable type for type pack");
  using TYPE = typename GetDataType<Ts...>::value;

  std::vector<std::string> vars{Ts::name()...};

  return typename SwarmPack<TYPE, Ts...>::Descriptor(
      static_cast<impl::SwarmPackDescriptor<TYPE>>(
          MakeSwarmPackDescriptor<TYPE>(swarm_name, vars)));
}
```
because the two `MakePackDescriptor` functions have different numbers of input variables, I *think* this should just work and should be unambiguous.

Reply:
ah yes! 😸 I think this is a viable path.  

I couldn't get the automatic deduction to work prior to introducing `MakeSwarmPackDescriptor` when trying to stay compatible with both string based a var-type based packs.  But now that this functionality is introduced, I think your path is a clear one to avoiding extraneous template arguments.

Reply:
This worked 😄